server/src/main/java/org/opensearch/common/cache/tier/TieredCache.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.tier;
+
+import org.opensearch.common.cache.ICache;
+import org.opensearch.common.cache.store.enums.CacheStoreType;
+
+/**
+ * This represents a cache comprising of multiple tiers/layers.
+ * @param <K> Type of key
+ * @param <V> Type of value
+ *
+ * @opensearch.experimental
+ */
+public interface TieredCache<K, V> extends ICache<K, V> {
+    Iterable<K> cacheKeys(CacheStoreType type);
+
+    void refresh(CacheStoreType type);
+}
```

Comment:
I am not inclined to exposing `CacheStoreType`, tiered cache should be fully managed and data access should be transparent irrespective of the tier data resides in

Reply:
Can we not create separate instance of on heap and disk cache where we configure intervals in the builder instantiation. Clean up is something internal to the cache management

Reply:
That would have been ideal.
But the way IndicesRequestCache is written made it  really tricky.
It essentially collects stale IndexReaders through CleanUpKey. As soon as any IndexReader goes stale, all keys referring to it will go stale as well. IndicesRequestCache then iterates over all the keys, see which have stale IndexReaders(through IndexReader.CacheKey), and delete it accordingly.

Let me know if you have a smart way to do that. I will also think more on this.
We also can address this in separate PR where we will adding some clean up logic related to TieredCache, let me know if this works too.

Reply:
The concern is there isn't a single integration that has been done with the IndicesRequestCache, which would have provided some validation of the interface you have chosen.

Reply:
We will planning to use ICache as an interface, which is extended by both StoreAwareCache/TieredCache . It has all the desired methods to integrate with IndicesRequestCache easily. As IndicesRequestCache should be agnostic of store type/tiered terminologies, it works fine.

As mentioned IndicesRequestCache has customized clean up logic and handles that logic on its own. With current setup TieredCache should also integrate seamlessly, just that we will performing cleanup for both onHeap/disk with a single thread.

Let me know if it makes sense.

Reply:
Regarding clean up logic, we can perform this even without being store aware. In my earlier point, we wanted to clean up onheap/disk separately for TieredCache. But there was always a way to clean up them together for which we don't need to expose Stores to consumer.

Reply:
>But there was always a way to clean up them together for which we don't need to expose Stores to consumer.

So you plan to remove it now?

Reply:
Removed it. :)